xcdat/spatial_avg.py

```diff
@@ -324,93 +325,155 @@ def _get_weights(
-
-                    # Scale the domain boundaries to the regional domain
-                    dom_bounds = self._scale_domain_to_region(dom_bounds, reg_bounds)
-
-                if dim == "lat":
-                    # The area between two lines of latitude scales as the
-                    # difference of the sine of latitude bounds. This scaling
-                    # is applied here.
-                    dom_bounds = np.sin(np.radians(dom_bounds))
-                    pmb_index = None
-
-                # Perform differencing to get weights and add metadata.
-                dim_wts = np.abs(dom_bounds[:, 1] - dom_bounds[:, 0])
-
-                # If pmb_index is not None it means that an extra
-                # longitude bound was added and the weight vector
-                # is too long. This extra weight should be added to the
-                # grid cell that crossed the prime meridian (and then
-                # removed from the dim_wts vector).
-                if pmb_index is not None:
-                    dim_wts[pmb_index] = dim_wts[pmb_index] + dim_wts[-1]
-                    dim_wts = dim_wts[0:-1]
-
-                # ensure attributes remain on dim_wts
-                dim_wts.attrs = dom_bounds.attrs
-                dim_wts.name = dim + "_wts"
-                axis_weights[dim] = dim_wts
-
-        weights = self._combine_weights(axis, axis_weights)
+            if dim in axis:
+                d_bounds = bounds_by_type["domain"]
+                self._validate_domain_bounds(d_bounds)
+
+                r_bounds = bounds_by_type["region"]
+                if r_bounds is not None:
+                    r_bounds = np.array(r_bounds, dtype="float")
+
+                if dim == "lon":
+                    weights = self._get_longitude_weights(d_bounds, r_bounds)
+                elif dim == "lat":
+                    weights = self._get_latitude_weights(d_bounds, r_bounds)
+
+                weights.attrs = d_bounds.attrs
+                weights.name = dim + "_wts"
+                axis_weights[dim] = weights
+
+        weights = self._combine_weights(axis_weights)
         return weights
 
-    def _align_longitude_to_360_axis(self, domain_bounds: xr.DataArray):
+    def _get_longitude_weights(
+        self, domain_bounds: xr.DataArray, region_bounds: np.array
+    ) -> xr.DataArray:
+        """Gets weights for the longitude axis.
+
+        This method performs longitudinal processing including (in order):
+
+        1. Align the axis orientations of the domain and region bounds to
+           (0, 360) to ensure compatibility in the proceeding steps.
+        2. Handle grid cells that cross the prime meridian (e.g., [-1, 1])
+           by recreating the axis with two additional grid cells ([0, 1] and
+           [359, 360]) to ensure alignment with the (0, 360) axis orientation.
+           The prime meridian grid cell is returned as a variable to handle
+           the length of weights in a proceeding step.
```

Comment:
Suggest rephrasing this: 

Handle grid cells that cross the prime meridian (e.g., [-1, 1]) by breaking such grid cells into two (e.g., [0, 1] and [359, 360]) to ensure alignment with the (0, 360) axis orientation. This results in a bounds axis of ``length(nlon)+1``. The index of the grid cell that crosses the prime meridian is returned in order to reduce the length of weights to ``nlon``.